### PR TITLE
libxl: Wait for argo-firewall-ready asynchronously

### PR DESCRIPTION
**Pull request by troyc**
Wait for argo-firewall-ready asynchronously using xswait

### CODE REVIEW

**Comment by eric-ch** [comment]
Could you move this logic out in its own patch?
`libxl-xenmgr-support.patch` conflates 3 changes mostly unrelated to each other:
- Disable QEMU for PV guest
- Have `shutdown_domain` to issue hard shutdown
- Wait for the Argo firewall rules to be written by `xenmgr`

File: recipes-extended/xen/files/libxl-xenmgr-support.patch [hunk 1]
```diff
@@ -108,31 +111,75 @@ PATCHES
...
++    if (ret) goto out;
+ 
+     return;
+ 
+@@ -2366,6 +2380,30 @@ out:
+     spawn_stubdom_pvqemu_cb(egc, &sdss->pvqemu, ret);
+ }
+ 
++static void argo_xswait_cb(libxl__egc *egc, libxl__xswait_state *xswait,
++                           int rc, const char *p)
```
Comment (jandryuk) [inline recipes-extended/xen/files/libxl-xenmgr-support.patch:157]: p here is the value of the watched xenstore node.  Do you want to verify it's the expected value?

**Review by jandryuk — COMMENTED** [review]
This generally looks reasonable.  However, there is no explanation in the commit message for *why* this change is being made.  Why convert to async?  Also, the timeout increase from 4 to 20 seconds is kinda hidden inside.  Why was that changed - is 4 seconds too short?  Please explain the reason for all changes as precisely as possible.

**Comment by troyc** [comment]
> Could you move this logic out in its own patch?
> `libxl-xenmgr-support.patch` conflates 3 changes mostly unrelated to each other:
> 
> * Disable QEMU for PV guest
> * Have `shutdown_domain` to issue hard shutdown
> * Wait for the Argo firewall rules to be written by `xenmgr`

Sure, I could split up the patch like this.

> This generally looks reasonable. However, there is no explanation in the commit message for _why_ this change is being made. Why convert to async? Also, the timeout increase from 4 to 20 seconds is kinda hidden inside. Why was that changed - is 4 seconds too short? Please explain the reason for all changes as precisely as possible.

The change is being made to preclude any issues caused by blocking and to better position the patches for upstreaming (depending on how the argo-viptables make their way upstream). There's no reason to change the timeout value, so it can remain at 4 seconds in my revision.

File: recipes-extended/xen/files/libxl-xenmgr-support.patch (hunk 1 above)
Comment (troyc) [reply recipes-extended/xen/files/libxl-xenmgr-support.patch:157]: Yes, I don't see any reason why not to verify p's value. However, I don't think there's anything better to do than to log an error message and continue on our merry way, same as in the case of a timeout or other error at this point.

**Comment by troyc** [comment]
I've applied the suggested changes to the branch. What's left in libxl-xenmgr-support.patch will become obsolete with vglass, with the possible exception of a few vif-related changes.

**Comment by troyc** [comment]
I updated the branch to add a commit message.

**Review by jandryuk — APPROVED** [review]
Thanks.

**Comment by eric-ch** [comment]
[Build](https://openxt.ainfosec.com/buildbot/#/builders/10/builds/36): [$mcq-36](https://openxt.ainfosec.com/OpenXT/auto/master/master-custom-quick/36/)

**Review by eric-ch — APPROVED** [review]

**Comment by eric-ch** [comment]
Will merge soon.

**Pull request merged**